File: src/main/java/by/iba/repair_report/service/ReportService.java
package by.iba.repair_report.service;

import by.iba.repair_report.dto.request.ReportRequest;
import by.iba.repair_report.dto.request.UpdateReportStatusRequest;
import by.iba.repair_report.dto.response.ReportResponse;
import by.iba.repair_report.dto.response.ReportSummaryResponse;

import java.time.YearMonth;
import java.util.List;

public interface ReportService {
    ReportResponse createReport(ReportRequest request);
    ReportResponse updateReport(Long reportId, ReportRequest request);
    ReportResponse updateReportStatus(Long reportId, UpdateReportStatusRequest request);
    void deleteReport(Long reportId);
    ReportResponse getReportById(Long reportId);
    List<ReportResponse> getReportsByBranch(Long branchId);
    List<ReportResponse> getReportsByBranchAndYear(Long branchId, Integer year);
    List<ReportResponse> getReportsByEnterprise(Long enterpriseId);
    List<ReportResponse> getReportsByAssociation(Long associationId);
    ReportResponse getReportByBranchAndPeriod(Long branchId, YearMonth period);
    ReportSummaryResponse getReportSummary(Long reportId);
    boolean existsReportForBranchAndPeriod(Long branchId, YearMonth period);
}

File: src/main/frontend/src/services/reportService.js
import api from './api';

export const reportService = {
    getAllReports: async () => {
        const response = await api.get('/reports');
        return response.data;
    },

    getReportById: async (id) => {
        const response = await api.get(`/reports/${id}`);
        return response.data;
    },

    createReport: async (reportData) => {
        const response = await api.post('/reports', reportData);
        return response.data;
    },

    updateReport: async (id, reportData) => {
        const response = await api.put(`/reports/${id}`, reportData);
        return response.data;
    },

    deleteReport: async (id) => {
        await api.delete(`/reports/${id}`);
    },

    getReportByBranchAndPeriod: async (branchId, period) => {
        const response = await api.get(`/reports/branch/${branchId}/period/${period}`);
        return response.data;
    },

    getReportsByBranchAndYear: async (branchId, year) => {
        const response = await api.get(`/reports/branch/${branchId}/year/${year}`);
        return response.data;
    },

    updateReportStatus: async (id, statusData) => {
        const response = await api.patch(`/reports/${id}/status`, statusData);
        return response.data;
    },

    getReportSummary: async (id) => {
        const response = await api.get(`/reports/${id}/summary`);
        return response.data;
    }
};